In [3]:
import pandas as pd 
from scipy.stats import pointbiserialr
import matplotlib.pyplot as plt

In [4]:
dataset = pd.read_csv("general_data.csv")

In [5]:
dataset

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [6]:
dataset.head()


Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [7]:
dataset.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [8]:
dataset.dropna()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404  Female  ...                 1.0      Y                18              8   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [9]:
dataset.Attrition.replace(to_replace = ('No','Yes'),value = (0,1),inplace = True)

In [15]:
stats, p = pearsonr(dataset.Attrition, dataset.Age)
print(stats,p)

-0.15920500686577965 1.996801615886744e-26


# Hypothesis 1

H0 = age & attrition has no correlation

H1 = age & attrition has correlation

r = -0.15920500686577965, i.e there is a negative correalation between age and attrition

p < 0.05, i.e. null hypothesis is rejected.

Hence we can conclude that there is a correalation between age and attrition because null hypothesis is rejected and the correlation between them is negative

In [17]:
stats, p = pearsonr(dataset.Attrition, dataset.MonthlyIncome)
print(stats,p)

-0.031176281698115007 0.03842748490600132


# Hypothesis 2

H0 = Monthly Income & attrition has no correlation

H1 = Monthly Income & attrition has correlation

r = -0.031176281698115007, i.e there is a negative correalation between MonthlyIncome and attrition

p < 0.05, i.e. null hypothesis is rejected.

Hence we can conclude that there is a correalation between age and attrition because null hypothesis is rejected and the correlation between them is negative

In [22]:
stats, p = pearsonr(dataset.Attrition, dataset.YearsWithCurrManager)
print(stats,p)

-0.15619931590162847 1.7339322652896276e-25


# Hypothesis 3

H0 =YearsWithCurrManager & attrition has no correlation 

H1 =YearsWithCurrManager & attrition has correlation

r = -0.15619931590162847, i.e there is a negative correalation between age and attrition

p < 0.05, i.e. null hypothesis is rejected.

Hence we can conclude that there is a correalation between YearsWithCurrManager and attrition because null hypothesis is rejected and the correlation between them is negative

In [23]:
stats, p = pearsonr(dataset.Attrition, dataset.DistanceFromHome)
print(stats,p)

-0.009730141010179674 0.5182860428050771


# Hypothesis 3

H0 =DistanceFromHome & attrition has no correlation

H1 =DistanceFromHome & attrition has correlation

r = -0.009730141010179674, i.e there is a negative correalation between age and attrition

p > 0.05, i.e. null hypothesis is accepted & alternate hypothesis is rejected.

Hence we can conclude that there is no correalation between DistanceFromHome and attrition.


In [24]:
stats, p = pearsonr(dataset.Attrition, dataset.YearsAtCompany)
print(stats,p)

-0.1343922139899772 3.1638831224877484e-19


# Hypothesis 4

H0 = Monthly Income & attrition has no correlation

H1 = Monthly Income & attrition has correlation

r = -0.1343922139899772, i.e there is a negative correalation between YearsAtCompany and attrition

p < 0.05, i.e. null hypothesis is rejected.

Hence we can conclude that there is a correalation between age and attrition because null hypothesis is rejected and the correlation between them is negative

In [28]:
stats, p = pearsonr(dataset.Attrition, dataset.Education)
print(stats,p)

-0.015111167710968713 0.3157293177118575


# Hypothesis 5

H0 =DistanceFromHome & attrition has no correlation

H1 =DistanceFromHome & attrition has correlation

r = -0.015111167710968713, i.e there is a negative correalation between age and attrition

p > 0.05, i.e. null hypothesis is accepted & alternate hypothesis is rejected.

Hence we can conclude that there is no correalation between Education and attrition.

In [29]:
stats, p = pearsonr(dataset.Attrition, dataset.PercentSalaryHike)
print(stats,p)

0.03253259489105349 0.030743386433355353


# Hypothesis 6

H0 =PercentSalaryHike & attrition has no correlation

H1 =PercentSalaryHike & attrition has correlation

r = 0.03253259489105349, i.e there is a positive correalation between PercentSalaryHike and attrition

p < 0.05, i.e. null hypothesis is rejected & alternate hypothesis is accepted

Hence we can conclude that there is positive correalation between PercentSalaryHike and attrition.

In [32]:
stats, p = pearsonr(dataset.Attrition, dataset.YearsAtCompany)
print(stats,p)

-0.1343922139899772 3.1638831224877484e-19


# Hypothesis 7

H0 =YearsAtCompany & attrition has no correlation

H1 =YearsAtCompany & attrition has correlation

r = -0.1343922139899772, i.e there is a negative correalation between YearsAtCompany and attrition

p < 0.05, i.e. null hypothesis is rejected & alternate hypothesis is accepted

Hence we can conclude that there is negative correalation between YearsAtCompany and attrition.


In [33]:
stats, p = pearsonr(dataset.Attrition, dataset.YearsSinceLastPromotion)
print(stats,p)

-0.03301877514258434 0.028330336189396753


# Hypothesis 8

H0 =YearsSinceLastPromotion & attrition has no correlation

H1 =YearsSinceLastPromotion & attrition has correlation

r = -0.03301877514258434, i.e there is a negative correalation between YearsSinceLastPromotion and attrition

p < 0.05, i.e. null hypothesis is rejected & alternate hypothesis is accepted

Hence we can conclude that there is negative correalation between YearsSinceLastPromotion and attrition.

In [35]:
stats, p = pearsonr(dataset.Attrition, dataset.TrainingTimesLastYear)
print(stats,p)

-0.049430576244255 0.0010247061915365072


# Hypothesis 9

H0 =TrainingTimesLastYear& attrition has no correlation

H1 =TrainingTimesLastYear & attrition has correlation

r = -0.049430576244255, i.e there is a negative correalation between TrainingTimesLastYear and attrition

p < 0.05, i.e. null hypothesis is rejected & alternate hypothesis is accepted

Hence we can conclude that there is negative correalation between TrainingTimesLastYear and attrition.

